From 2696ce7c9a12d053a7b9fae067d6aacd470025ff Mon Sep 17 00:00:00 2001
From: Bhavana Jupalli <Bhavana.Jupalli@amd.com>
Date: Wed, 2 Apr 2025 14:51:46 +0530
Subject: [PATCH] pciepsu: Fix GCC compilation warning

Fix GCC compilation warning
"variable 'MaxBarSize' set but not used [-Wunused-but-set-variable]"

Signed-off-by: Bhavana Jupalli <bhavana.jupalli@amd.com>
Acked-by: Bharat Kumar Gogada <bharat.kumar.gogada@amd.com>
Upstream-Status: Backport
---
 .../drivers/pciepsu/src/xpciepsu.c                 | 14 ++++++++------
 .../drivers/pciepsu/src/xpciepsu_g.c               |  3 ++-
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu.c b/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu.c
index 1e9abe7c83..4cfe6c0cfd 100644
--- a/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu.c
+++ b/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu.c
@@ -773,15 +773,10 @@ static void XPciePsu_BarMemoryAlloc(XPciePsu *InstancePtr, u8 Bus,u8 Device,u8 F
 #endif
 		} else {
 
+			/* 32 bit AS is required */
 			MaxBarSize = InstancePtr->Config.NpMemMaxAddr - InstancePtr->Config.NpMemBaseAddr;
 			TestWrite = XPciePsu_PositionRightmostSetbit(Size[BarNo]);
 			Value[BarNo] = 2 << (TestWrite-1);
-
-#if defined(__aarch64__) || defined(__arch64__)
-
-			/* 32 bit AS is required */
-			MemAs = Size[BarNo];
-
 			if (BarAllocControl != 0) {
 				if(Value[BarNo] > MaxBarSize) {
 
@@ -791,7 +786,14 @@ static void XPciePsu_BarMemoryAlloc(XPciePsu *InstancePtr, u8 Bus,u8 Device,u8 F
 						(Value[BarNo] / 1024),Bus,Device,Function);
 					return;
 				}
+			}
+
+#if defined(__aarch64__) || defined(__arch64__)
 
+			/* 32 bit AS is required */
+			MemAs = Size[BarNo];
+
+			if (BarAllocControl != 0) {
 			/* actual bar size is 2 << TestWrite */
 			BarAddr =
 				XPciePsu_ReserveBarMem(
diff --git a/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu_g.c b/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu_g.c
index f12be28dd2..222b97a5b4 100644
--- a/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu_g.c
+++ b/XilinxProcessorIPLib/drivers/pciepsu/src/xpciepsu_g.c
@@ -1,6 +1,6 @@
 /******************************************************************************
 * Copyright (C) 2018 - 2020 Xilinx, Inc.  All rights reserved.
-* Copyright (C) 2022 - 2023 Advanced Micro Devices, Inc.  All rights reserved.
+* Copyright (C) 2022 - 2025 Advanced Micro Devices, Inc.  All rights reserved.
 * SPDX-License-Identifier: MIT
 ******************************************************************************/
 
@@ -22,6 +22,7 @@ XPciePsu_Config XPciePsu_ConfigTable[] =
 		XPAR_PSU_PCIE_HIGH1_BASEADDR,
 		XPAR_PSU_PCIE_LOW_HIGHADDR,
 		XPAR_PSU_PCIE_HIGH1_HIGHADDR,
+		XPAR_PSU_PCIE_DMA_BASEADDR,
 		XPAR_PSU_PCIE_PCIE_MODE
 	}
 };
-- 
2.47.2

